### ISSUE TITLE
Flaky E2E: Help Doc article is shown

### ISSUE THREAD

**Issue by worldomonation** [Automattic/wp-calypso#78968]
### Spec file

specs/help-center/help-center__general-interaction.ts

### TeamCity ID

10321966

### Logs

![image](https://github.com/Automattic/wp-calypso/assets/6549265/5daaf62b-3913-46a7-832f-f8ed3fb521f2)

```
======= Failed test run #1 ==========
  locator.waitFor: Timeout 10000ms exceeded.
  =========================== logs ===========================
  waiting for locator('.help-center__container').getByRole('article').getByRole('heading').first() to be visible
  ============================================================
  at SupportComponent.waitFor [as getOpenArticleTitle] (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/packages/calypso-e2e/src/lib/components/support-component.ts:106:17)
      at Object.getOpenArticleTitle (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/test/e2e/specs/help-center/help-center__general-interaction.ts:42:48)
      at Promise.then.completed (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:293:28)
      at callAsyncCircusFn (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:226:10)
      at _callCircusTest (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:297:40)
      at _runTest (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:233:3)
      at _runTestsForDescribeBlock (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:135:9)
      at _runTestsForDescribeBlock (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:130:9)
      at _runTestsForDescribeBlock (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:130:9)
      at run (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:68:3)
      at runAndTransformResultsToJestFormat (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)
      at jestAdapter (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)
      at runTestInternal (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:367:16)
      at runTest (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:444:34)
      at Object.worker (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/testWorker.js:106:12)
======= Failed test run #2 ==========
  locator.waitFor: Timeout 10000ms exceeded.
  =========================== logs ===========================
  waiting for locator('.help-center__container').getByRole('article').getByRole('heading').first() to be visible
  ============================================================
  at SupportComponent.waitFor [as getOpenArticleTitle] (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/packages/calypso-e2e/src/lib/components/support-component.ts:106:17)
      at Object.getOpenArticleTitle (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/test/e2e/specs/help-center/help-center__general-interaction.ts:42:48)
      at Promise.then.completed (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:293:28)
      at callAsyncCircusFn (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:226:10)
      at _callCircusTest (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:297:40)
      at _runTest (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:233:3)
      at _runTestsForDescribeBlock (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:135:9)
      at _runTestsForDescribeBlock (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:130:9)
      at _runTestsForDescribeBlock (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:130:9)
      at run (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:68:3)
      at runAndTransformResultsToJestFormat (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)
      at jestAdapter (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)
      at runTestInternal (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:367:16)
      at runTest (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:444:34)
```

**Comment by worldomonation** [comment]
![image](https://github.com/Automattic/wp-calypso/assets/6549265/1107a34f-f980-436c-a818-432acf230ebf)

[help-center__general-interaction__help-doc-article-is-shown__2023-07-04_09-54-50__1-1.webm](https://github.com/Automattic/wp-calypso/assets/6549265/f5ac3836-12b0-45ec-a6ae-3029db008574)

**Comment by worldomonation** [comment]
Sister issue: `Calypso Link opens in a new page`

![image](https://github.com/Automattic/wp-calypso/assets/6549265/390166bc-aa04-4b94-a131-04c58bf153e2)

```
page.waitForEvent: Timeout 10000ms exceeded while waiting for event "popup"
=========================== logs ===========================
waiting for event "popup"
============================================================
at Object.<anonymous> (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/test/e2e/specs/help-center/help-center__general-interaction.ts:65:15)
    at Promise.then.completed (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:391:28)
    at callAsyncCircusFn (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:316:10)
    at _callCircusTest (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:218:40)
    at _runTest (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:155:3)
    at _runTestsForDescribeBlock (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:66:9)
    at _runTestsForDescribeBlock (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:60:9)
    at _runTestsForDescribeBlock (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:60:9)
    at run (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:25:3)
    at runAndTransformResultsToJestFormat (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)
    at jestAdapter (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:82:19)
    at runTestInternal (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:389:16)
    at runTest (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:475:34)
    at Object.worker (/home/teamcity-3/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/testWorker.js:133:12)
```

**Comment by dpasque** [comment]
Re-opening this issue because it's back 😬 

Did some interesting debugging, and it's actually an auth failure on the help document request!

So, this request:
`https://public-api.wordpress.com/wpcom/v2/help/article/9619154/138080?_envelope=1`

Returns this payload:
```
{
  "body": {
    "code": "unauthorized",
    "message": "User cannot view post",
    "data": 403
  },
  "status": 500,
  "headers": {
    "Allow": "GET"
  }
}
```

This really doesn't seem right to me... Why would we be blocking a logged in user from help articles?? Going to ping some folks to figure this out. 

And actually, looking at this more specifically, this seems very focused on that specific article! You can reproduce this with any non super-admin -- just search for Account Closure and try to select it, it won't work! But you can open the support article directly? 

I'm going to open an issue. In the meantime, we can also make the test exclude the article and move on if the `Account Closure` is the name of the article. That's janky, but maybe better than a mute?

Thoughts @worldomonation?

**Comment by dpasque** [comment]
Issue opened and triaged here: #79576.

It's low priority, and AFAIK, it only affects the one article, so I'm thinking maybe we just try a search that steers clear of this article!

**Comment by dpasque** [comment]
Fixed by #79580
